### ISSUE TITLE
TypeError: socket.destroy is not a function

### ISSUE THREAD

**Issue by gajus** [TooTallNate/node-agent-base#32]
Started to get error:

```
TypeError: socket.destroy is not a function
    at freeSocket (/srv/node_modules/agent-base/index.js:169:10)
    at EventEmitter.onfree (/srv/node_modules/agent-base/index.js:138:9)
    at EventEmitter.emit (events.js:210:5)
    at EventEmitter.emit (domain.js:476:20)
    at onSocketNT (_http_client.js:720:14)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)

```

**Comment by gajus** [comment]
Sounds related to https://github.com/TooTallNate/node-https-proxy-agent/issues/81

**Comment by gajus** [comment]
For what it is worth, I am using:

```
"http-proxy-agent": "^2.1.0",
"https-proxy-agent": "^2.2.3",

```

**Comment by gajus** [comment]
Does not appear when using https-proxy-agent@3.0.1.

**Comment by TooTallNate** [comment]
Fixed in https-proxy-agent@2.2.4 as well.
